packages/assistify-help-request/server/publications/Expertise.js

```diff
@@ -17,23 +17,25 @@ Meteor.publish('autocompleteExpertise', function(selector) {
 		},
 		limit: 10
 	};
-
-	const cursorHandle = RocketChat.models.Rooms.findByNameContainingTypesAndTags(selector.term, [{type: 'e'}], options)
-		.observeChanges({
-			added(_id, record) {
-				return pub.added('autocompleteRecords', _id, record);
-			},
-			changed(_id, record) {
-				return pub.changed('autocompleteRecords', _id, record);
-			},
-			removed(_id, record) {
-				return pub.removed('autocompleteRecords', _id, record);
-			}
-		});
+	let cursorHandle = RocketChat.models.Rooms.findByNameContainingTypesAndTags(selector.term, [{type: 'e'}], options);
+	if (RocketChat.models.Rooms.find({t: 'e'}).count() < 10 && cursorHandle.count() === 0) {
```

Comment:
This logic (return all if less than 10) should not be handled inside the publication.
Instead, the consumer should check the count and then execute the subscription without criteria

Reply:
alternatively, the semantic of this publication should be "ExpertiseWithMinimumCount" (and thus have two parameters, the second being the count which is optional)

Reply:
will look into it and see it if I can adapt the logic as mentioned.